src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -215,10 +215,10 @@ internal Binder GetSpeculativeBinder(int position, ExpressionSyntax expression,
             }
 
             // May be binding an expression in a context that doesn't have a LocalScopeBinder in the chain.
-            return new LocalScopeBinder(binder);
+            return new ExpressionVariableBinder(expression, binder);
```

Comment:
How would you suggest I test for the difference between your suggestion and the implementation here?